### ISSUE TITLE
Upgrade 9.1.6.2 ->  10.0.2.1 null cache when moving avatars outside home (fed shares)

### ISSUE THREAD

**Issue by XHAGBARD** [owncloud/core#28763]
### Steps to reproduce
1. apply repository for 10.0 version
2. yum upgrade and install new version
3. Apply upgrade from the Web and error ocurred
4. Apply upgrade via occ cli and debug:

```
[root@owncloud ~]# sudo -u apache /usr/bin/php /var/www/html/owncloud/occ maintenance:mode --off && sudo -u apache /usr/bin/php /var/www/html/owncloud/occ upgrade -vvv
ownCloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Maintenance mode disabled
ownCloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
2017-08-22T09:43:33+02:00 Set log level to debug
2017-08-22T09:43:33+02:00 Turned on maintenance mode
2017-08-22T09:43:33+02:00 Repair step: Repair MySQL database engine
2017-08-22T09:43:33+02:00 Repair step: Repair MySQL collation
2017-08-22T09:43:33+02:00 Repair info: All tables already have the correct collation -> nothing to do
2017-08-22T09:43:33+02:00 Repair step: Repair SQLite autoincrement
2017-08-22T09:43:33+02:00 Repair step: Repair duplicate entries in oc_lucene_status
2017-08-22T09:43:33+02:00 Repair info: lucene_status table does not exist -> nothing to do
2017-08-22T09:43:33+02:00 Repair step: Upgrade app code from the marketplace
2017-08-22T09:43:33+02:00 Repair info: Enabling market app to assist with update
2017-08-22T09:43:33+02:00 Repair info: Using market to update existing apps
2017-08-22T09:43:33+02:00 Repair info: Attempting to update the following existing compatible apps from market: 
2017-08-22T09:43:33+02:00 Repair info: Fetching app from market: activity
2017-08-22T09:43:34+02:00 Repair info: 
2017-08-22T09:43:34+02:00 Repair info: Fetching app from market: calendar
2017-08-22T09:43:34+02:00 Repair info: 
2017-08-22T09:43:34+02:00 Repair info: Fetching app from market: comments
2017-08-22T09:43:35+02:00 Repair info: App (comments) is not known at the marketplace.
2017-08-22T09:43:35+02:00 Repair info: Fetching app from market: configreport
2017-08-22T09:43:35+02:00 Repair info: App (configreport) is not known at the marketplace.
2017-08-22T09:43:35+02:00 Repair info: Fetching app from market: contacts
2017-08-22T09:43:35+02:00 Repair info: 
2017-08-22T09:43:35+02:00 Repair info: Fetching app from market: dav
2017-08-22T09:43:36+02:00 Repair info: App (dav) is not known at the marketplace.
2017-08-22T09:43:36+02:00 Repair info: Fetching app from market: encryption
2017-08-22T09:43:36+02:00 Repair info: App (encryption) is not known at the marketplace.
2017-08-22T09:43:36+02:00 Repair info: Fetching app from market: external
2017-08-22T09:43:36+02:00 Repair info: App (external) is not known at the marketplace.
2017-08-22T09:43:36+02:00 Repair info: Fetching app from market: federatedfilesharing
2017-08-22T09:43:37+02:00 Repair info: App (federatedfilesharing) is not known at the marketplace.
2017-08-22T09:43:37+02:00 Repair info: Fetching app from market: federation
2017-08-22T09:43:37+02:00 Repair info: App (federation) is not known at the marketplace.
2017-08-22T09:43:37+02:00 Repair info: Fetching app from market: files
2017-08-22T09:43:38+02:00 Repair info: App (files) is not known at the marketplace.
2017-08-22T09:43:38+02:00 Repair info: Fetching app from market: files_external
2017-08-22T09:43:38+02:00 Repair info: App (files_external) is not known at the marketplace.
2017-08-22T09:43:38+02:00 Repair info: Fetching app from market: files_pdfviewer
2017-08-22T09:43:38+02:00 Repair info: 
2017-08-22T09:43:38+02:00 Repair info: Fetching app from market: files_sharing
2017-08-22T09:43:39+02:00 Repair info: App (files_sharing) is not known at the marketplace.
2017-08-22T09:43:39+02:00 Repair info: Fetching app from market: files_texteditor
2017-08-22T09:43:39+02:00 Repair info: 
2017-08-22T09:43:39+02:00 Repair info: Fetching app from market: files_trashbin
2017-08-22T09:43:39+02:00 Repair info: App (files_trashbin) is not known at the marketplace.
2017-08-22T09:43:39+02:00 Repair info: Fetching app from market: files_versions
2017-08-22T09:43:40+02:00 Repair info: App (files_versions) is not known at the marketplace.
2017-08-22T09:43:40+02:00 Repair info: Fetching app from market: files_videoplayer
2017-08-22T09:43:40+02:00 Repair info: App (files_videoplayer) is not known at the marketplace.
2017-08-22T09:43:40+02:00 Repair info: Fetching app from market: firstrunwizard
2017-08-22T09:43:41+02:00 Repair info: App (firstrunwizard) is not known at the marketplace.
2017-08-22T09:43:41+02:00 Repair info: Fetching app from market: gallery
2017-08-22T09:43:41+02:00 Repair info: 
2017-08-22T09:43:41+02:00 Repair info: Fetching app from market: market
2017-08-22T09:43:41+02:00 Repair info: 
2017-08-22T09:43:41+02:00 Repair info: Fetching app from market: notifications
2017-08-22T09:43:42+02:00 Repair info: App (notifications) is not known at the marketplace.
2017-08-22T09:43:42+02:00 Repair info: Fetching app from market: ownbackup
2017-08-22T09:43:42+02:00 Repair info: 
2017-08-22T09:43:42+02:00 Repair info: Fetching app from market: provisioning_api
2017-08-22T09:43:42+02:00 Repair info: App (provisioning_api) is not known at the marketplace.
2017-08-22T09:43:42+02:00 Repair info: Fetching app from market: templateeditor
2017-08-22T09:43:43+02:00 Repair info: App (templateeditor) is not known at the marketplace.
2017-08-22T09:43:43+02:00 Repair info: Fetching app from market: updatenotification
2017-08-22T09:43:43+02:00 Repair info: App (updatenotification) is not known at the marketplace.
2017-08-22T09:43:43+02:00 Repair info: App was not updated: activity
2017-08-22T09:43:43+02:00 Repair info: App was not updated: calendar
2017-08-22T09:43:43+02:00 Repair info: App was not updated: comments
2017-08-22T09:43:43+02:00 Repair info: App was not updated: configreport
2017-08-22T09:43:43+02:00 Repair info: App was not updated: contacts
2017-08-22T09:43:43+02:00 Repair info: App was not updated: dav
2017-08-22T09:43:43+02:00 Repair info: App was not updated: encryption
2017-08-22T09:43:43+02:00 Repair info: App was not updated: external
2017-08-22T09:43:43+02:00 Repair info: App was not updated: federatedfilesharing
2017-08-22T09:43:43+02:00 Repair info: App was not updated: federation
2017-08-22T09:43:43+02:00 Repair info: App was not updated: files
2017-08-22T09:43:43+02:00 Repair info: App was not updated: files_external
2017-08-22T09:43:43+02:00 Repair info: App was not updated: files_pdfviewer
2017-08-22T09:43:43+02:00 Repair info: App was not updated: files_sharing
2017-08-22T09:43:43+02:00 Repair info: App was not updated: files_texteditor
2017-08-22T09:43:43+02:00 Repair info: App was not updated: files_trashbin
2017-08-22T09:43:43+02:00 Repair info: App was not updated: files_versions
2017-08-22T09:43:43+02:00 Repair info: App was not updated: files_videoplayer
2017-08-22T09:43:43+02:00 Repair info: App was not updated: firstrunwizard
2017-08-22T09:43:43+02:00 Repair info: App was not updated: gallery
2017-08-22T09:43:43+02:00 Repair info: App was not updated: market
2017-08-22T09:43:43+02:00 Repair info: App was not updated: notifications
2017-08-22T09:43:43+02:00 Repair info: App was not updated: ownbackup
2017-08-22T09:43:43+02:00 Repair info: App was not updated: provisioning_api
2017-08-22T09:43:43+02:00 Repair info: App was not updated: templateeditor
2017-08-22T09:43:43+02:00 Repair info: App was not updated: updatenotification
2017-08-22T09:43:43+02:00 Updating database schema
2017-08-22T09:43:43+02:00 Updated database
2017-08-22T09:43:43+02:00 Repair step: Repair mime types
2017-08-22T09:43:43+02:00 Repair step: Generate ETags for file where no ETag is present.
2017-08-22T09:43:45+02:00 Repair info: ETags have been fixed for 0 files/folders.
2017-08-22T09:43:45+02:00 Repair step: Clean tags and favorites
2017-08-22T09:43:45+02:00 Repair info: 0 tags of deleted users have been removed.
2017-08-22T09:43:45+02:00 Repair info: 0 tags for delete files have been removed.
2017-08-22T09:43:45+02:00 Repair info: 0 tag entries for deleted tags have been removed.
2017-08-22T09:43:45+02:00 Repair info: 0 tags with no entries have been removed.
2017-08-22T09:43:45+02:00 Repair step: Drop old database tables
2017-08-22T09:43:45+02:00 Drop old database tables
2017-08-22T09:43:45+02:00 
                                                    2017-08-22T09:43:45+02:00  Done
 28/28 [============================] 100%2017-08-22T09:43:45+02:00 
2017-08-22T09:43:45+02:00 Repair step: Drop old background jobs
2017-08-22T09:43:45+02:00 Repair step: Remove getetag entries in properties table
2017-08-22T09:43:45+02:00 Repair info: Removed 0 unneeded "{DAV:}getetag" entries from properties table.
2017-08-22T09:43:45+02:00 Repair step: Repair outdated OCS IDs
2017-08-22T09:43:45+02:00 Repair step: Repair invalid shares
2017-08-22T09:43:45+02:00 Repair step: Remove old share propagation app entries
2017-08-22T09:43:45+02:00 Repair step: Fix permissions so avatars can be stored again
2017-08-22T09:43:45+02:00 Fix permissions so avatars can be stored again
                                                    2017-08-22T09:43:47+02:00  Done
 2/2 [============================] 100%2017-08-22T09:43:47+02:00 
2017-08-22T09:43:47+02:00 Repair step: Move user avatars outside the homes to the new location
2017-08-22T09:43:47+02:00 Move user avatars outside the homes to the new location
                                                    2017-08-22T09:43:47+02:00  Starting ...
  0/40 [>-]   0%2017-08-22T09:43:47+02:00 :47+02:00 
2017-08-22T09:43:47+02:00                                                                           
2017-08-22T09:43:47+02:00   [Symfony\Component\Debug\Exception\FatalThrowableError]                 
2017-08-22T09:43:47+02:00   Call to a member function getCache() on null  
2017-08-22T09:43:47+02:00                                                                           

2017-08-22T09:43:47+02:00 Exception trace:
2017-08-22T09:43:47+02:00  () at /var/www/html/owncloud/lib/private/Files/Mount/MountPoint.php:251
2017-08-22T09:43:47+02:00  OC\Files\Mount\MountPoint->getStorageRootId() at /var/www/html/owncloud/lib/private/Files/Config/UserMountCache.php:92
2017-08-22T09:43:47+02:00  OC\Files\Config\UserMountCache->OC\Files\Config\{closure}() at n/a:n/a
2017-08-22T09:43:47+02:00  array_map() at /var/www/html/owncloud/lib/private/Files/Config/UserMountCache.php:97
2017-08-22T09:43:47+02:00  OC\Files\Config\UserMountCache->registerMounts() at /var/www/html/owncloud/lib/private/Files/Config/MountProviderCollection.php:134
2017-08-22T09:43:47+02:00  OC\Files\Config\MountProviderCollection->registerMounts() at /var/www/html/owncloud/lib/private/Files/Filesystem.php:448
2017-08-22T09:43:47+02:00  OC\Files\Filesystem::initMountPoints() at /var/www/html/owncloud/apps/files_sharing/lib/SharedStorage.php:102
2017-08-22T09:43:47+02:00  OCA\Files_Sharing\SharedStorage->init() at /var/www/html/owncloud/apps/files_sharing/lib/SharedStorage.php:455
2017-08-22T09:43:47+02:00  OCA\Files_Sharing\SharedStorage->getWrapperStorage() at /var/www/html/owncloud/lib/private/Files/Storage/Wrapper/Wrapper.php:490
2017-08-22T09:43:47+02:00  OC\Files\Storage\Wrapper\Wrapper->instanceOfStorage() at /var/www/html/owncloud/apps/files_sharing/lib/SharedStorage.php:129
2017-08-22T09:43:47+02:00  OCA\Files_Sharing\SharedStorage->instanceOfStorage() at /var/www/html/owncloud/lib/private/legacy/util.php:158
2017-08-22T09:43:47+02:00  OC_Util::{closure}() at /var/www/html/owncloud/lib/private/Files/Storage/StorageFactory.php:100
2017-08-22T09:43:47+02:00  OC\Files\Storage\StorageFactory->wrap() at /var/www/html/owncloud/lib/private/Files/Storage/StorageFactory.php:82
2017-08-22T09:43:47+02:00  OC\Files\Storage\StorageFactory->getInstance() at /var/www/html/owncloud/lib/private/Files/Mount/MountPoint.php:138
2017-08-22T09:43:47+02:00  OC\Files\Mount\MountPoint->createStorage() at /var/www/html/owncloud/lib/private/Files/Mount/MountPoint.php:161
2017-08-22T09:43:47+02:00  OC\Files\Mount\MountPoint->getStorage() at /var/www/html/owncloud/lib/private/Files/View.php:1362
2017-08-22T09:43:47+02:00  OC\Files\View->getFileInfo() at /var/www/html/owncloud/lib/private/Files/Node/Root.php:182
2017-08-22T09:43:47+02:00  OC\Files\Node\Root->get() at n/a:n/a
2017-08-22T09:43:47+02:00  call_user_func_array() at /var/www/html/owncloud/lib/private/Files/Node/LazyRoot.php:66
2017-08-22T09:43:47+02:00  OC\Files\Node\LazyRoot->__call() at /var/www/html/owncloud/lib/private/Files/Node/LazyRoot.php:143
2017-08-22T09:43:47+02:00  OC\Files\Node\LazyRoot->get() at /var/www/html/owncloud/lib/private/Repair/MoveAvatarOutsideHome.php:111
2017-08-22T09:43:47+02:00  OC\Repair\MoveAvatarOutsideHome->moveAvatars() at /var/www/html/owncloud/lib/private/Repair/MoveAvatarOutsideHome.php:135
2017-08-22T09:43:47+02:00  OC\Repair\MoveAvatarOutsideHome->OC\Repair\{closure}() at /var/www/html/owncloud/lib/private/User/Manager.php:378
2017-08-22T09:43:47+02:00  OC\User\Manager->OC\User\{closure}() at /var/www/html/owncloud/lib/private/User/AccountMapper.php:220
2017-08-22T09:43:47+02:00  OC\User\AccountMapper->callForAllUsers() at /var/www/html/owncloud/lib/private/User/Manager.php:379
2017-08-22T09:43:47+02:00  OC\User\Manager->callForAllUsers() at /var/www/html/owncloud/lib/private/User/Manager.php:397
2017-08-22T09:43:47+02:00  OC\User\Manager->callForSeenUsers() at /var/www/html/owncloud/lib/private/Repair/MoveAvatarOutsideHome.php:141
2017-08-22T09:43:47+02:00  OC\Repair\MoveAvatarOutsideHome->run() at /var/www/html/owncloud/lib/private/Repair.php:88
2017-08-22T09:43:47+02:00  OC\Repair->run() at /var/www/html/owncloud/lib/private/Updater.php:250
2017-08-22T09:43:47+02:00  OC\Updater->doUpgrade() at /var/www/html/owncloud/lib/private/Updater.php:108
2017-08-22T09:43:47+02:00  OC\Updater->upgrade() at /var/www/html/owncloud/core/Command/Upgrade.php:249
2017-08-22T09:43:47+02:00  OC\Core\Command\Upgrade->execute() at /var/www/html/owncloud/lib/composer/symfony/console/Command/Command.php:262
2017-08-22T09:43:47+02:00  Symfony\Component\Console\Command\Command->run() at /var/www/html/owncloud/lib/composer/symfony/console/Application.php:826
2017-08-22T09:43:47+02:00  Symfony\Component\Console\Application->doRunCommand() at /var/www/html/owncloud/lib/composer/symfony/console/Application.php:189
2017-08-22T09:43:47+02:00  Symfony\Component\Console\Application->doRun() at /var/www/html/owncloud/lib/composer/symfony/console/Application.php:120
2017-08-22T09:43:47+02:00  Symfony\Component\Console\Application->run() at /var/www/html/owncloud/lib/private/Console/Application.php:160
2017-08-22T09:43:47+02:00  OC\Console\Application->run() at /var/www/html/owncloud/console.php:99
2017-08-22T09:43:47+02:00  require_once() at /var/www/html/owncloud/occ:11
2017-08-22T09:43:47+02:00 
2017-08-22T09:43:47+02:00 upgrade [--no-app-disable]
2017-08-22T09:43:47+02:00 
```

### Expected behaviour
No upgrade finish

### Actual behaviour
Tell us what happens instead

### Server configuration
**Operating system**: Centos 7.3.1611

**Web server:** httpd

**Database:** Percona Server 5.7.18-16

**PHP version:** 7.0.22

**ownCloud version:** 10.0.2.1

**Updated from an older ownCloud or fresh install:** 9.1.6.2

**Where did you install ownCloud from:** /var/www/html/owncloud

**Signing status (ownCloud 9.0 and above):**

**The content of config/config.php:**

```
$CONFIG = array (
  'updatechecker' => false,
  'instanceid' => 'ockn8ye244ke',
  'passwordsalt' => 'xxxxxxxxxxxxxxxxxxxxxxxx',
  'secret' => '**************************************',
  'trusted_domains' =>
  array (
    0 => 'xx.xx.xx.xx',
    1 => 'xxxxxxxxxxxxxx',
  ),
  'datadirectory' => '/var/www/html/owncloud/data',
  'overwrite.cli.url' => 'https:/xxxxxxxxxxxxxxxxxxxxx',
  'dbtype' => 'mysql',
  'version' => '9.1.6.2',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'xxxxxxxxx'',
  'dbpassword' => 'xxxxxxxxx',
  'logtimezone' => 'Europe/Madrid',
  'installed' => true,
  'default_language' => 'es',
  'mail_from_address' => 'no-email',
  'mail_smtpmode' => 'php',
  'mail_domain' => 'airenetworks.es',
  'appstore.experimental.enabled' => true,
  'theme' => '',
  'maintenance' => true,
  'loglevel' => 0,
  'debug' => true,
  'trashbin_retention_obligation' => 'auto',
  'singleuser' => false,
);
```

**List of activated apps:**

```
Enabled:
  - activity: 2.3.4
  - calendar: 1.4.2
  - comments: 0.3.0
  - configreport: 0.1.1
  - contacts: 1.5.2
  - dav: 0.2.9
  - encryption: 1.3.0
  - external: 1.2
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.0
  - files_pdfviewer: 0.8.2
  - files_sharing: 0.10.0
  - files_texteditor: 2.2
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1Enabled:
  - activity: 2.3.4
  - calendar: 1.4.2
  - comments: 0.3.0
  - configreport: 0.1.1
  - contacts: 1.5.2
  - dav: 0.2.9
  - encryption: 1.3.0
  - external: 1.2
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.0
  - files_pdfviewer: 0.8.2
  - files_sharing: 0.10.0
  - files_texteditor: 2.2
  - files_trashbin: 0.9.0
  - notifications: 0.3.0
  - ownbackup: 17.7.0
  - provisioning_api: 0.5.0
  - templateeditor: 0.1
  - updatenotification: 0.2.1
Disabled:
  - announcementcenter
  - documents
  - files_antivirus
  - html5_videoplayer
  - notes
  - onlyoffice
  - ownnote
  - qownnotesapi
  - registration
  - richdocuments
  - systemtags
  - tasks
  - theme-example
  - user_external
```

**Are you using encryption:** yes

#### ownCloud log

```
{"reqId":"JotOqRfp48nXftw81DLt","level":0,"time":"2017-08-22T09:43:45+02:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"--","message":"str_repeat(): Second argument has to be greater than or equal to 0 at \/var\/www\/html\/owncloud\/lib\/composer\/symfony\/console\/Helper\/ProgressBar.php#517"}
{"reqId":"JotOqRfp48nXftw81DLt","level":0,"time":"2017-08-22T09:43:46+02:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"--","message":"str_repeat(): Second argument has to be greater than or equal to 0 at \/var\/www\/html\/owncloud\/lib\/composer\/symfony\/console\/Helper\/ProgressBar.php#517"}
{"reqId":"JotOqRfp48nXftw81DLt","level":3,"time":"2017-08-22T09:43:47+02:00","remoteAddr":"","user":"--","app":"core","method":"--","url":"--","message":"cURL error 51: Unable to communicate securely with peer: requested domain name does not match the server's certificate."}
{"reqId":"GT5qGzrpQq0MQt53VCoH","level":0,"time":"2017-08-22T09:45:01+02:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Update required, skipping cron"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":0,"time":"2017-08-22T09:50:56+02:00","remoteAddr":"88.148.82.4","user":"--","app":"core","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"starting upgrade from 9.1.6.2 to 10.0.2.1"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":0,"time":"2017-08-22T09:51:05+02:00","remoteAddr":"88.148.82.4","user":"--","app":"PHP","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"Cannot modify header information - headers already sent at \/var\/www\/html\/owncloud\/lib\/private\/User\/Session.php#861"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":0,"time":"2017-08-22T09:51:05+02:00","remoteAddr":"88.148.82.4","user":"--","app":"PHP","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"Cannot modify header information - headers already sent at \/var\/www\/html\/owncloud\/lib\/private\/User\/Session.php#862"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":0,"time":"2017-08-22T09:51:05+02:00","remoteAddr":"88.148.82.4","user":"--","app":"PHP","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"Cannot modify header information - headers already sent at \/var\/www\/html\/owncloud\/lib\/private\/User\/Session.php#863"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":0,"time":"2017-08-22T09:51:05+02:00","remoteAddr":"88.148.82.4","user":"--","app":"PHP","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"Cannot modify header information - headers already sent at \/var\/www\/html\/owncloud\/lib\/private\/User\/Session.php#866"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":0,"time":"2017-08-22T09:51:05+02:00","remoteAddr":"88.148.82.4","user":"--","app":"PHP","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"Cannot modify header information - headers already sent at \/var\/www\/html\/owncloud\/lib\/private\/User\/Session.php#867"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":0,"time":"2017-08-22T09:51:05+02:00","remoteAddr":"88.148.82.4","user":"--","app":"PHP","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"Cannot modify header information - headers already sent at \/var\/www\/html\/owncloud\/lib\/private\/User\/Session.php#868"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":0,"time":"2017-08-22T09:51:05+02:00","remoteAddr":"88.148.82.4","user":"--","app":"PHP","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"session_regenerate_id(): Cannot regenerate session id - headers already sent at \/var\/www\/html\/owncloud\/lib\/private\/Session\/Internal.php#114"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":0,"time":"2017-08-22T09:51:05+02:00","remoteAddr":"88.148.82.4","user":"--","app":"PHP","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"A session had already been started - ignoring session_start() at \/var\/www\/html\/owncloud\/lib\/private\/Session\/Internal.php#98"}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":3,"time":"2017-08-22T09:51:09+02:00","remoteAddr":"88.148.82.4","user":"--","app":"core","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"cURL error 51: Unable to communicate securely with peer: requested domain name does not match the server's certificate."}
{"reqId":"WZviX1IxbT4DrvEneyit4QAAAAY","level":3,"time":"2017-08-22T09:51:09+02:00","remoteAddr":"88.148.82.4","user":"--","app":"PHP","method":"GET","url":"\/core\/ajax\/update.php?requesttoken=WDJYLHUAFgMiIDN8KgIqTTggFAd6HQZNFn8YHXAtJAk%3D%3AibwkDJrpXrf3pjy%2FHvBP4OtbN%2BonGaeJX7MBvXtTov4%3D","message":"Error: Call to a member function getCache() on null at \/var\/www\/html\/owncloud\/lib\/private\/Files\/Mount\/MountPoint.php#251"}
```

**Comment by XHAGBARD** [comment]
I disable app files_sharing and it work now....

**Comment by XHAGBARD** [comment]
If active the app, error via Web:

```
Tipo: Error
Código: 0
Mensaje: Call to a member function getCache() on null
Archivo: /var/www/html/owncloud/lib/private/Files/Mount/MountPoint.php
Línea: 251

Trazas

#0 /var/www/html/owncloud/lib/private/Files/Config/UserMountCache.php(92): OC\Files\Mount\MountPoint->getStorageRootId()
#1 [internal function]: OC\Files\Config\UserMountCache->OC\Files\Config\{closure}(Object(OCA\Files_Sharing\External\Mount))
#2 /var/www/html/owncloud/lib/private/Files/Config/UserMountCache.php(97): array_map(Object(Closure), Array)
#3 /var/www/html/owncloud/lib/private/Files/Config/MountProviderCollection.php(134): OC\Files\Config\UserMountCache->registerMounts(Object(OC\User\User), Array)
#4 /var/www/html/owncloud/lib/private/Files/Filesystem.php(448): OC\Files\Config\MountProviderCollection->registerMounts(Object(OC\User\User), Array)
#5 /var/www/html/owncloud/lib/private/Files/Filesystem.php(375): OC\Files\Filesystem::initMountPoints('j.peral')
#6 /var/www/html/owncloud/lib/private/legacy/util.php(297): OC\Files\Filesystem::init('j.peral', '/j.peral/files')
#7 /var/www/html/owncloud/lib/private/Files/Filesystem.php(353): OC_Util::setupFS()
#8 /var/www/html/owncloud/lib/private/Files/View.php(1114): OC\Files\Filesystem::resolvePath('/files_encrypti...')
#9 /var/www/html/owncloud/lib/private/Files/View.php(524): OC\Files\View->basicOperation('file_exists', '/files_encrypti...')
#10 /var/www/html/owncloud/lib/private/Encryption/Keys/Storage.php(215): OC\Files\View->file_exists('/files_encrypti...')
#11 /var/www/html/owncloud/lib/private/Encryption/Keys/Storage.php(112): OC\Encryption\Keys\Storage->getKey('/files_encrypti...')
#12 /var/www/html/owncloud/apps/encryption/lib/KeyManager.php(569): OC\Encryption\Keys\Storage->getSystemUserKey('pubShare_005b51...', 'OC_DEFAULT_MODU...')
#13 /var/www/html/owncloud/apps/encryption/lib/KeyManager.php(157): OCA\Encryption\KeyManager->getPublicShareKey()
#14 /var/www/html/owncloud/apps/encryption/lib/Users/Setup.php(85): OCA\Encryption\KeyManager->validateShareKey()
#15 /var/www/html/owncloud/apps/encryption/lib/AppInfo/Application.php(72): OCA\Encryption\Users\Setup->setupSystem()
#16 /var/www/html/owncloud/apps/encryption/appinfo/app.php(30): OCA\Encryption\AppInfo\Application->__construct(Array, true)
#17 /var/www/html/owncloud/lib/private/legacy/app.php(213): require_once('/var/www/html/o...')
#18 /var/www/html/owncloud/lib/private/legacy/app.php(164): OC_App::requireAppFile('encryption')
#19 /var/www/html/owncloud/lib/private/legacy/app.php(121): OC_App::loadApp('encryption')
#20 /var/www/html/owncloud/lib/base.php(902): OC_App::loadApps()
#21 /var/www/html/owncloud/index.php(49): OC::handleRequest()
#22 {main}
```

**Comment by PVince81** [comment]
Can you try `select * from oc_storages` where id like 'local::/%` ?

I wonder if the "root" storage entry is broken in some way.

**Comment by PVince81** [comment]
Also check if the folder "data/avatars" exists now, considering that the avatar migration failed.

cc @butonic

**Comment by XHAGBARD** [comment]
```
mysql> select * from oc_storages where id like 'local::/%';
+-------------------------------------+------------+-----------+--------------+
| id                                  | numeric_id | available | last_checked |
+-------------------------------------+------------+-----------+--------------+
| local::/var/www/html/owncloud/data/ |          2 |         1 |         NULL |
+-------------------------------------+------------+-----------+--------------+
```

**Comment by XHAGBARD** [comment]
This is normal? An entry in oc_filecache without patch:

```
mysql> select * from oc_filecache where storage=2 and fileid=10;
+--------+---------+------+----------------------------------+--------+------+----------+----------+--------+------------+---------------+-----------+------------------+---------------+-------------+----------+
| fileid | storage | path | path_hash                        | parent | name | mimetype | mimepart | size   | mtime      | storage_mtime | encrypted | unencrypted_size | etag          | permissions | checksum |
+--------+---------+------+----------------------------------+--------+------+----------+----------+--------+------------+---------------+-----------+------------------+---------------+-------------+----------+
|     10 |       2 |      | d41d8cd98f00b204e9800998ecf8427e |     -1 |      |        2 |        1 | 393348 | 1503391188 |    1503389580 |         0 |                0 | 599bedd42337e |          23 | NULL     |
+--------+---------+------+----------------------------------+--------+------+----------+----------+--------+------------+---------------+-----------+------------------+---------------+-------------+----------+
```

**Comment by PVince81** [comment]
the root entry has no path, yes this is normal.

**Comment by XHAGBARD** [comment]
Annotation.

When you disable the files_sharing app you can complete the upgrade correctly. But when you re-enable it (with the app updated) it still fails with the same error message.

**Comment by kswtch** [comment]
@XHAGBARD
I had the same issue like you. After upgrading to version 10 of owncloud I couldn't log in anymore and the same error messages as you posted appeared. As you explained disabling the files_sharing app solves the problem so it had to do something with the sharing part. I found out that we both had the following error message in our debug log:
> "cURL error 51: Unable to communicate securely with peer: requested domain name does not match the server's certificate."

So I checked all remote shares in the database:
`select * from oc_share_external;`
This will give you a list of all remote shares. You can visit the URLs with your browser and check for invalid SSL certificates. Delete the ones with invalid certificates:
`delete from oc_share_external where id=23 limit 1;`

I also removed the shares to the storage with invalid certificates from the table _oc_share_.
After that everything works as expected. 

Maybe in some future release an invalid SSL certificate of a remote share does not break owncloud but disables the share automatically and puts a warning in the logs (or to the user).

**Comment by pep29** [comment]
I also have the same Upgrade issue except that i am upgrading from 9.1.6.2 to 10.0.3.3

**Comment by pep29** [comment]
I issued the new ticket here: https://github.com/owncloud/core/issues/29209

if this is not neccessary, please close it

**Comment by PVince81** [comment]
### Steps to reproduce the stack trace (not upgrade)

1. Setup an OC server on https with self-signed certificate: "https://localhost" (this would be the sharer of federated share). Let's call this server B. (I'm not even sure you need to have an actual OC there)
2. Setup a local OC server A
3. On A, insert this entry into the database: `insert into oc_share_external (id, remote, share_token, password, name, owner, user, mountpoint, mountpoint_hash, accepted, remote_id) values (1,'https://localhost/','e6t6z1F96fizfCN',NULL,'/Photos','admin','admin','/Photos','24c87f459b360e13ce71d229de4f6ce6',1,3);`
4. Log in in the web UI of A
5. Check ownCloud log:

```
{\"Exception\":\"Error\",\"Message\":\"Call to a member function getCache() on null\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Config\\\/UserMountCache.php(92): OC\\\\Files\\\\Mount\\\\MountPoint->getStorageRootId()\\n#1 [internal function]: OC\\\\Files\\\\Config\\\\UserMountCache->OC\\\\Files\\\\Config\\\\{closure}(Object(OCA\\\\Files_Sharing\\\\External\\\\Mount))\\n#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Config\\\/UserMountCache.php(97): array_map(Object(Closure), Array)\\n#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Config\\\/MountProviderCollection.php(134): OC\\\\Files\\\\Config\\\\UserMountCache->registerMounts(Object(OC\\\\User\\\\User), Array)\\n#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Filesystem.php(448): OC\\\\Files\\\\Config\\\\MountProviderCollection->registerMounts(Object(OC\\\\User\\\\User), Array)\\n#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Cache\\\/File.php(52): OC\\\\Files\\\\Filesystem::initMountPoints('admin')\\n#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Cache\\\/File.php(173): OC\\\\Cache\\\\File->getStorage()\\n#7 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/base.php(737): OC\\\\Cache\\\\File->gc()\\n#8 [internal function]: OC::{closure}(Object(OC\\\\User\\\\User), 'admin')\\n#9 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Hooks\\\/EmitterTrait.php(99): call_user_func_array(Object(Closure), Array)\\n#10 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Hooks\\\/PublicEmitter.php(33): OC\\\\Hooks\\\\BasicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postLogin', Array)\\n#11 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/User\\\/Session.php(477): OC\\\\Hooks\\\\PublicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postLogin', Array)\\n#12 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/User\\\/Session.php(301): OC\\\\User\\\\Session->loginWithPassword(*** sensitive parameters replaced ***)\\n#13 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/core\\\/Controller\\\/LoginController.php(201): OC\\\\User\\\\Session->login(*** sensitive parameters replaced ***)\\n#14 [internal function]: OC\\\\Core\\\\Controller\\\\LoginController->tryLogin(*** sensitive parameters replaced ***)\\n#15 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(159): call_user_func_array(Array, Array)\\n#16 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(89): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OC\\\\Core\\\\Controller\\\\LoginController), 'tryLogin')\\n#17 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(98): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OC\\\\Core\\\\Controller\\\\LoginController), 'tryLogin')\\n#18 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(46): OC\\\\AppFramework\\\\App::main('LoginController', 'tryLogin', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#19 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#20 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(307): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#21 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/base.php(928): OC\\\\Route\\\\Router->match('\\\/login')\\n#22 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/index.php(56): OC::handleRequest()\\n#23 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Mount\\\/MountPoint.php\",\"Line\":251}"}
```

That should be enough to debug this.

**Comment by pep29** [comment]
Both A and B should be own 9.1.6.2?

**Comment by PVince81** [comment]
This is happening because the storage class for the federated share has a programming mistake: the constructor is not supposed to initiate any connection to the remote. The connection needs to be initiated only once an actual access is requested.

The error is happening here: https://github.com/owncloud/core/blob/v10.0.3/apps/files_sharing/lib/External/Storage.php#L82. This code is trying to find out the remote endpoints by connecting to the remote's "ocs-provider" endpoint. This is happening too early.

The sad thing about this is that not doing this call makes it impossible to have a value for the "root".

Now we could try to catch the exception in question. We're already catching ClientException and ConnectException. Unfortunately for this one case, Guzzle throws a `RingException` which is thrown by Ringphp, it does not extend any of the Guzzle base exceptions... This also means that other code paths could be affected by the same uncaught exception.

**Comment by PVince81** [comment]
There are two fixes to work on:

- [ ] Mitigate the issue by handling null storages in `MountPoint->getStorageRootId()` and return -1
- [ ] Find a way to properly catch RingException in all federated share related code paths
- [ ] Prevent the constructor of the federated share external storage class to do any remote connections and do one lazily when any of the access methods is used: failure to do so will likely slow down a lot of operations when no memcache was set.

The mitigation already makes the exception disappear for me.

**Comment by PVince81** [comment]
@pep29 in my case both are 10.0.3. My reproduction case is not related to an upgrade, it's just a different way of reproducing the same core issue.

I expect that the fixes will also help repairing the upgrades.

@pep29 If you still have a setup where you can reproduce this, can you try with the following patch before upgrading: https://github.com/owncloud/core/pull/29210 (download as patch: https://patch-diff.githubusercontent.com/raw/owncloud/core/pull/29210.patch)

**Comment by PVince81** [comment]
@pep29 in your case you need to apply the patch on the 10.0.3 code, just before running `occ upgrade`

**Comment by pep29** [comment]
I do not use external storage, which is why the "oc_share_external" is empty.

Can you please tell me more about the Mitigate, i could try it, I mean how exactly?

Since i upgraded after disabling the files_sharing app, i guess it is already to late to use the patch #29210

**Comment by pep29** [comment]
@PVinc281 Unfortunately the patch #29210 does not help me, i stuck get stuck in "Move user avatars outside the homes to the new location" step when the files_sharing is enabled

2017-10-11T15:55:48+00:00 Set log level to debug
2017-10-11T15:55:48+00:00 Repair step: Repair MySQL database engine
2017-10-11T15:55:48+00:00 Repair step: Repair MySQL collation
2017-10-11T15:55:48+00:00 Repair info: All tables already have the correct collation -> nothing to do
2017-10-11T15:55:48+00:00 Repair step: Repair SQLite autoincrement
2017-10-11T15:55:48+00:00 Repair step: Repair duplicate entries in oc_lucene_status
2017-10-11T15:55:48+00:00 Repair info: removing duplicate entries from lucene_status
2017-10-11T15:55:48+00:00 Repair step: Upgrade app code from the marketplace
2017-10-11T15:55:48+00:00 Repair info: Enabling market app to assist with update
2017-10-11T15:55:48+00:00 Repair info: Using market to update existing apps
2017-10-11T15:55:48+00:00 Repair info: Attempting to update the following missing apps from market: activity, files_pdfviewer, files_texteditor, gallery
2017-10-11T15:55:48+00:00 Repair info: Fetching app from market: activity
2017-10-11T15:55:50+00:00 Repair info: Fetching app from market: files_pdfviewer
2017-10-11T15:55:52+00:00 Repair info: Fetching app from market: files_texteditor
2017-10-11T15:55:54+00:00 Repair info: Fetching app from market: gallery
2017-10-11T15:55:55+00:00 Repair info: Attempting to update the following existing compatible apps from market: activity, files_pdfviewer, files_texteditor, gallery
2017-10-11T15:55:55+00:00 Repair info: Fetching app from market: comments
2017-10-11T15:55:56+00:00 Repair info: App (comments) is not known at the marketplace.
2017-10-11T15:55:56+00:00 Repair info: Fetching app from market: configreport
2017-10-11T15:55:56+00:00 Repair info: App (configreport) is not known at the marketplace.
2017-10-11T15:55:56+00:00 Repair info: Fetching app from market: dav
2017-10-11T15:55:56+00:00 Repair info: App (dav) is not known at the marketplace.
2017-10-11T15:55:56+00:00 Repair info: Fetching app from market: federatedfilesharing
2017-10-11T15:55:56+00:00 Repair info: App (federatedfilesharing) is not known at the marketplace.
2017-10-11T15:55:56+00:00 Repair info: Fetching app from market: federation
2017-10-11T15:55:57+00:00 Repair info: App (federation) is not known at the marketplace.
2017-10-11T15:55:57+00:00 Repair info: Fetching app from market: files
2017-10-11T15:55:57+00:00 Repair info: App (files) is not known at the marketplace.
2017-10-11T15:55:57+00:00 Repair info: Fetching app from market: files_external
2017-10-11T15:55:57+00:00 Repair info: App (files_external) is not known at the marketplace.
2017-10-11T15:55:57+00:00 Repair info: Fetching app from market: files_sharing
2017-10-11T15:55:58+00:00 Repair info: App (files_sharing) is not known at the marketplace.
2017-10-11T15:55:58+00:00 Repair info: Fetching app from market: files_trashbin
2017-10-11T15:55:58+00:00 Repair info: App (files_trashbin) is not known at the marketplace.
2017-10-11T15:55:58+00:00 Repair info: Fetching app from market: files_versions
2017-10-11T15:55:58+00:00 Repair info: App (files_versions) is not known at the marketplace.
2017-10-11T15:55:58+00:00 Repair info: Fetching app from market: files_videoplayer
2017-10-11T15:55:59+00:00 Repair info: App (files_videoplayer) is not known at the marketplace.
2017-10-11T15:55:59+00:00 Repair info: Fetching app from market: firstrunwizard
2017-10-11T15:55:59+00:00 Repair info: App (firstrunwizard) is not known at the marketplace.
2017-10-11T15:55:59+00:00 Repair info: Fetching app from market: market
2017-10-11T15:55:59+00:00 Repair info: 
2017-10-11T15:55:59+00:00 Repair info: Fetching app from market: notifications
2017-10-11T15:56:00+00:00 Repair info: App (notifications) is not known at the marketplace.
2017-10-11T15:56:00+00:00 Repair info: Fetching app from market: provisioning_api
2017-10-11T15:56:00+00:00 Repair info: App (provisioning_api) is not known at the marketplace.
2017-10-11T15:56:00+00:00 Repair info: Fetching app from market: systemtags
2017-10-11T15:56:00+00:00 Repair info: App (systemtags) is not known at the marketplace.
2017-10-11T15:56:00+00:00 Repair info: Fetching app from market: updatenotification
2017-10-11T15:56:01+00:00 Repair info: App (updatenotification) is not known at the marketplace.
2017-10-11T15:56:01+00:00 Repair info: App was not updated: comments
2017-10-11T15:56:01+00:00 Repair info: App was not updated: configreport
2017-10-11T15:56:01+00:00 Repair info: App was not updated: dav
2017-10-11T15:56:01+00:00 Repair info: App was not updated: federatedfilesharing
2017-10-11T15:56:01+00:00 Repair info: App was not updated: federation
2017-10-11T15:56:01+00:00 Repair info: App was not updated: files
2017-10-11T15:56:01+00:00 Repair info: App was not updated: files_external
2017-10-11T15:56:01+00:00 Repair info: App was not updated: files_sharing
2017-10-11T15:56:01+00:00 Repair info: App was not updated: files_trashbin
2017-10-11T15:56:01+00:00 Repair info: App was not updated: files_versions
2017-10-11T15:56:01+00:00 Repair info: App was not updated: files_videoplayer
2017-10-11T15:56:01+00:00 Repair info: App was not updated: firstrunwizard
2017-10-11T15:56:01+00:00 Repair info: App was not updated: market
2017-10-11T15:56:01+00:00 Repair info: App was not updated: notifications
2017-10-11T15:56:01+00:00 Repair info: App was not updated: provisioning_api
2017-10-11T15:56:01+00:00 Repair info: App was not updated: systemtags
2017-10-11T15:56:01+00:00 Repair info: App was not updated: updatenotification
2017-10-11T15:56:01+00:00 Updating database schema
2017-10-11T15:56:01+00:00 Updated database
2017-10-11T15:56:01+00:00 Repair step: Repair mime types
2017-10-11T15:56:01+00:00 Repair step: Generate ETags for file where no ETag is present.
2017-10-11T15:56:01+00:00 Repair info: ETags have been fixed for 0 files/folders.
2017-10-11T15:56:01+00:00 Repair step: Clean tags and favorites
2017-10-11T15:56:01+00:00 Repair info: 0 tags of deleted users have been removed.
2017-10-11T15:56:01+00:00 Repair info: 0 tags for delete files have been removed.
2017-10-11T15:56:01+00:00 Repair info: 0 tag entries for deleted tags have been removed.
2017-10-11T15:56:01+00:00 Repair info: 0 tags with no entries have been removed.
2017-10-11T15:56:01+00:00 Repair step: Drop old database tables
2017-10-11T15:56:01+00:00 Drop old database tables
2017-10-11T15:56:01+00:00 
                                                    2017-10-11T15:56:01+00:00  Done
 28/28 [============================] 100%2017-10-11T15:56:01+00:00 
2017-10-11T15:56:01+00:00 Repair step: Drop old background jobs
2017-10-11T15:56:01+00:00 Repair step: Remove getetag entries in properties table
2017-10-11T15:56:01+00:00 Repair info: Removed 0 unneeded "{DAV:}getetag" entries from properties table.
2017-10-11T15:56:01+00:00 Repair step: Repair outdated OCS IDs
2017-10-11T15:56:01+00:00 Repair step: Repair invalid shares
2017-10-11T15:56:01+00:00 Repair step: Remove old share propagation app entries
2017-10-11T15:56:01+00:00 Repair step: Fix permissions so avatars can be stored again
2017-10-11T15:56:01+00:00 Fix permissions so avatars can be stored again
                                                    2017-10-11T15:56:02+00:00  Done
 2/2 [============================] 100%2017-10-11T15:56:02+00:00 
2017-10-11T15:56:02+00:00 Repair step: Move user avatars outside the homes to the new location
2017-10-11T15:56:02+00:00 Move user avatars outside the homes to the new location
                                                    2017-10-11T15:56:02+00:00  Starting ...
  0/56 [>-]   0%          2017-10-11T15:56:02+00:00

**Comment by pep29** [comment]
In the log file, i got this 
{"reqId":"k5wJcNsxFQh2AZ0i2OWH","level":3,"time":"2017-10-11T15:56:02+00:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"--","message":"str_repeat(): Second argument has to be greater than or equal to 0 at \/srv\/www\/owncloud\/lib\/composer\/symfony\/console\/Helper\/ProgressBar.php#517"}

**Comment by PVince81** [comment]
@pep29 thanks for testing. Seems the stack trace is gone.
Does it hang there or just crash ?

If it hangs, I suspect it might be trying to connect to federated share servers which are not available and is waiting for a connection timeout. If that's the case, this would fall under the third category to fix: "Prevent the constructor of the federated share external storage class to do any remote connections"

**Comment by pep29** [comment]
@PVince81 It just hang there till it timeout. Since i do not use external storage, can i get ride of it?

Another thing, i use two different owncloud for two different websites A and B, the upgrade on B went smootless, however it not that heavily used as A. Could be the huge amount of Data on A be a problem?

**Comment by PVince81** [comment]
@pep29 but you are using federated shares or some of your users do ? Federated shares internally work like external storages. If there aren't that many, you could check the contents of the "oc_share_external" table. I suspect that maybe some of the domains in that list are not reacahable any more.

The quantity of data in the ownClouds should not matter for this one specific spot of the upgrade.

**Comment by pep29** [comment]
@PVince81 the "oc_share_external" table is indeed empty for both mentioned site A and B

**Comment by PVince81** [comment]
@pep29 uh oh... ok, so it means it's yet another bug. It is possible that my patch above mitigated it a bit.

If it's neither external storage nor incoming federated shares, then it can likely only be somewhat broken local shares.

@pep29 mind filling the issue template and posting it as a comment here ? I'd like to see what apps you have enabled for more clues.

**Comment by pep29** [comment]
@PVince81 We indeed have some problems with local share (many folder are for some reasons duplicated with the extra [2]), there also many tickets about this duplicate issue.

here is a list of the used apps (i also use files_sharing, but for now it is disabled)
Enabled:
  - activity: 2.3.6
  - comments: 0.3.0
  - configreport: 0.1.1
  - dav: 0.3.0
  - federatedfilesharing: 0.3.1
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.1
  - files_pdfviewer: 0.8.2
  - files_texteditor: 2.2.1
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 16.0.2
  - market: 0.2.2
  - notifications: 0.3.1
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - updatenotification: 0.2.1
Disabled:
  - encryption
  - external
  - files_antivirus
  - files_sharing
  - templateeditor
  - theme-example
  - user_external

**Comment by PVince81** [comment]
@pep29 let's continue in https://github.com/owncloud/core/issues/29209 then, focussing on the broken local shares

**Comment by PVince81** [comment]
PR for lazily connecting to federated shares, should improve update speed: https://github.com/owncloud/core/pull/29275 (it shouldn't hang for such a long time!)

**Comment by PVince81** [comment]
Two tasks from https://github.com/owncloud/core/issues/28763#issuecomment-335851401 done.

Remaining one is about catching RingException

**Comment by PVince81** [comment]
I have retested the 9.1->10.0.4beta2 upgrade scenario. With the two fixes in place, there is no more leaking of RingException and the upgrade properly goes through.
